src/main/kotlin/com/aliucord/gradle/task/CompileDexTask.kt

```diff
@@ -48,72 +47,74 @@ abstract class CompileDexTask : DefaultTask() {
-                        dexOutputDir.toPath()
-                    )
+                dexBuilder.convert(
+                    files.stream(),
+                    dexOutputDir.toPath(),
+                    null
+                )
 
-                    for (file in files) {
-                        val reader = ClassReader(file.readAllBytes())
+                for (file in files) {
+                    val reader = ClassReader(file.readAllBytes())
 
-                        val classNode = ClassNode()
-                        reader.accept(classNode, 0)
+                    val classNode = ClassNode()
+                    reader.accept(classNode, 0)
 
-                        for (annotation in classNode.visibleAnnotations.orEmpty() + classNode.invisibleAnnotations.orEmpty()) {
-                            if (annotation.desc == "Lcom/aliucord/annotations/AliucordPlugin;") {
-                                val aliucord = project.extensions.getAliucord()
+                    for (annotation in classNode.visibleAnnotations.orEmpty() + classNode.invisibleAnnotations.orEmpty()) {
+                        if (annotation.desc == "Lcom/aliucord/annotations/AliucordPlugin;") {
+                            val aliucord = project.extensions.getAliucord()
 
-                                require(aliucord.pluginClassName == null) {
-                                    "Only 1 active plugin class per project is supported"
-                                }
+                            require(aliucord.pluginClassName == null) {
+                                "Only 1 active plugin class per project is supported"
+                            }
 
-                                for (method in classNode.methods) {
-                                    if (method.name == "getManifest" && method.desc == "()Lcom/aliucord/entities/Plugin\$Manifest;") {
-                                        throw IllegalArgumentException("Plugin class cannot override getManifest, use manifest.json system!")
-                                    }
+                            for (method in classNode.methods) {
+                                if (method.name == "getManifest" && method.desc == "()Lcom/aliucord/entities/Plugin\$Manifest;") {
+                                    throw IllegalArgumentException("Plugin class cannot override getManifest, use manifest.json system!")
                                 }
-
-                                aliucord.pluginClassName = classNode.name.replace('/', '.')
-                                    .also { pluginClassFile.asFile.orNull?.writeText(it) }
                             }
+
+                            aliucord.pluginClassName = classNode.name.replace('/', '.')
+                                .also { pluginClassFile.asFile.orNull?.writeText(it) }
                         }
                     }
                 }
-        }
+            }
+
+        bootClassPath.close()
+        classPath.close()
```

Comment:
This won't close them if an error occurs, you should go back to using `Closer`

I don't really see how the changes to this class improve anything other than a bit of formatting

Reply:
Can I just put a `try catch finally`
I don't see the point of a whole library for 2 close statements

Reply:
oh i didn't realize it was a whole ass library
yeah just use a finally block